tests/geobase/checks_FairGeoBasicShape.h

```diff
@@ -0,0 +1,162 @@
+/********************************************************************************
+ *   Copyright (C) 2024 GSI Helmholtzzentrum fuer Schwerionenforschung GmbH     *
+ *                                                                              *
+ *              This software is distributed under the terms of the             *
+ *              GNU Lesser General Public Licence (LGPL) version 3,             *
+ *                  copied verbatim in the file "LICENSE"                       *
+ ********************************************************************************/
+
+#include "FairGeoBasicShape.h"
+#include "FairGeoRotation.h"
+#include "FairGeoTransform.h"
+#include "FairGeoVector.h"
+#include "FairGeoVolume.h"
+#include "generate_TestInput.h"
+
+#include <TArrayD.h>
+#include <catch2/catch_test_macros.hpp>
+#include <catch2/matchers/catch_matchers_floating_point.hpp>
+#include <cstdio>
+#include <string>
+
+using Catch::Matchers::WithinRel;
+
+#ifndef TESTS_GEOBASE_CHECKS_FAIRGEOBASICSHAPE_H
+#define TESTS_GEOBASE_CHECKS_FAIRGEOBASICSHAPE_H
+
+namespace fairroot::tests
+{
+
+auto check_FairGeoTransform(FairGeoTransform& _trans)
+{
+    const FairGeoRotation& rot = _trans.getRotMatrix();
+    REQUIRE(rot.isUnitMatrix());
+
+    const FairGeoVector& vec = _trans.getTransVector();
+    REQUIRE_THAT(vec.getValues(0), WithinRel(0., 0.001));
+    REQUIRE_THAT(vec.getValues(1), WithinRel(0., 0.001));
+    REQUIRE_THAT(vec.getValues(2), WithinRel(0., 0.001));
+
+    const FairGeoVector& vec_cm = _trans.getTranslation();
+    REQUIRE_THAT(vec_cm.getValues(0), WithinRel(0., 0.001));
+    REQUIRE_THAT(vec_cm.getValues(1), WithinRel(0., 0.001));
+    REQUIRE_THAT(vec_cm.getValues(2), WithinRel(0., 0.001));
+}
+
+auto check_FairGeoShape_Construction(FairGeoBasicShape& shape,
+                                     std::string name,
+                                     int numPoints,
+                                     int numParams,
+                                     TArrayD* _param)
+{
+
+    SECTION("construction")
+    {
+        // Check that members are correctly set by constructor
+        REQUIRE(shape.GetName() == name);
+        REQUIRE(shape.getNumPoints() == numPoints);
+        REQUIRE(shape.getNumParam() == numParams);
+
+        TArrayD* param = shape.getParam();
+        if (!_param) {
+            REQUIRE(param == _param);
+        } else {
+            REQUIRE(param->GetSize() == _param->GetSize());
+        }
```

Comment:
How is param_size connected to the TArrayD? I don't get the syntax.

Reply:
In the old code you perform two checks:

1. ```cpp
        if (!_param) {
            REQUIRE(param == _param);
   ```

   This basically checks that if param is a nullptr that your expected value (`_param`) is also a nullptr.

2. ```cpp
            REQUIRE(param->GetSize() == _param->GetSize());
   ```

   Here you only compare the sizes of both TArrayDs. Nothing more.

So you care for two properties of `param`:
1. It being nullptr or not-nullptr
2. It having the correct size.

This can well be represented by a `std::optional<Int_t>` (`Int_t`, because that's the return value of GetSize).
* If the std::optional has no value: we expect `param == nullptr`
* If the std::optional has a value, then we expect the size of `param` to be the value of the optional.

Reply:
After looking at the change again I understood the issue. I misinterpreted the std::optional as local variable and not as function parameter.

Reply:
I changed the code such that now also the parameters are checked.